src/mscorlib/src/System/Buffer.cs

```diff
@@ -545,47 +538,62 @@ internal unsafe static void Memmove(byte* dest, byte* src, nuint len)
 #if BIT64
                 *(long*)(dest + i) = *(long*)(src + i);
                 *(long*)(dest + i + 8) = *(long*)(src + i + 8);
-#else
+#else // BIT64
                 *(int*)(dest + i) = *(int*)(src + i);
                 *(int*)(dest + i + 4) = *(int*)(src + i + 4);
                 *(int*)(dest + i + 8) = *(int*)(src + i + 8);
                 *(int*)(dest + i + 12) = *(int*)(src + i + 12);
-#endif
+#endif // BIT64
 
                 i = counter;
-                
+
                 // See notes above for why this wasn't used instead
                 // i += 16;
             }
             while (counter <= end);
 
-            if ((len & 8) != 0)
+            // We have <= 15 bytes to copy at this point.
+
+            if ((mask & 8) != 0)
```

Comment:
Consider, `if (mask == 0) { return; }`, otherwise we are doing four pointless comparisons